### PR TITLE
Add site gallery. Fix #1084

### LINKED ISSUE #1084: Showcase Gallery
Our gallery was originally designed for debugging and thus it shows the whole canvas for all visualizations.  This takes up a lot of space. 

However, showing just clipped examples would make the layout much nicer.  
For example, protovis's gallery looks really nice. 
https://mbostock.github.io/protovis/ex/

protovis_-_examples

### PR DESCRIPTION
**Pull request by ZeningQu**
- Add `gallery.md` page with "Showcase", "Basic", "Stacked" and "Trellis" sections. The section names correspond to group names in `vl-examples.json`. `gallery.md` specifies sections, `vl-examples.json` populates specific examples, using svgs generated by `scripts/generate-images.sh`. `vl-examples.json` also specifies custom css properties for certain examples.
- Examples link to vega-lite editor.
- The gallery page should look like this (some examples don't show up because we are missing the svgs): screencapture-127-0-0-1-4000-vega-lite-gallery-html-1455760814636

### CODE REVIEW

File: site/static/main.css [hunk 1]
```diff
@@ -239,6 +239,37 @@ nav a:hover {
...
+  width: calc(100%/4 - 32px);
+  margin: 16px 32px 16px 0px;
+}
+.gallery .image {
+  display: block;
+  /* Make the height depend on the width */
+  width: 100%;
+  /* hack to make height depend on width*/
+  padding-bottom: 100%;
+  border: 1px dotted #CECECE;
```
Comment (domoritz) [inline site/static/main.css]: can you make the border 2px and solid?

File: site/static/main.css [hunk 2]
```diff
@@ -239,6 +239,37 @@ nav a:hover {
...
+  background-repeat: no-repeat;
+  background-clip: padding-box;
+}
+.gallery .image:hover {
+  border: 1px dashed orange;
```
Comment (domoritz) [inline site/static/main.css]: Let's make the border darker, rather than orange

File: site/static/main.css [hunk 3]
```diff
@@ -239,6 +239,37 @@ nav a:hover {
   background-repeat: no-repeat;
 }
 
+/* Gallery */
+
+.gallery {
```
Comment (domoritz) [inline site/static/main.css]: can you also add some code to not underline the links on hover. I think it looks nicer

File: site/static/showcasegallery.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+d3.json('examples/vl-examples.json', function(error, VL_SPECS) {
```
Comment (domoritz) [inline site/static/showcasegallery.ts]: Let's move all of this to main.ts

File: examples/vl-examples.json [hunk 5]
```diff
@@ -3,16 +3,28 @@
     {
       "name": "bar_layered_transparent",
       "title": "Bar chart with transparent layers",
-      "showInEditor": true
+      "showInEditor": true,
```
Comment (domoritz) [inline examples/vl-examples.json]: Feel free to delete all of the "showInEditor" entries. The examples that have showInGallery:false can be deleted.

**Comment by domoritz** [comment]
Fantastic. Looks really good! I made a few minor comments but it's pretty much ready to go after you address those.

File: examples/vl-examples.json (hunk 5 above)
Comment (kanitw) [inline examples/vl-examples.json]: Deleted from this listing file (but not from the specs folder)

**Comment by ZeningQu** [comment]
I've made those changes! Please feel free to comment / merge.

**Comment by kanitw** [comment]
ok, I will try to modify it as I see fits.

**Comment by kanitw** [comment]
Ok I have just modified the vl-examples.json.    It would be great if you could optimize the gallery parameters for:

- Scatter Plot with Log Scale
- Area Chart 
- Table Heatmap
- Stacked Area Chart
- Normalized Stacked Area Chart 
- Streamgraph
- Trellis Scatter Plot

**Comment by ZeningQu** [comment]
I updated gallery parameters for some of the examples, but I can't see Stacked Area Chart, Normalized Stacked Area Chart, and Streamgraph. How did you get the svgs for these examples?

**Comment by kanitw** [comment]
@domoritz  any idea?

**Pull request merged**